### PR TITLE
rank/teamrank: Display how much better you are

### PR DESCRIPTION
**Pull request by def-**
screenshot-20210123@231642

## Checklist

- [x] Tested the change ingame
- [x] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [x] Considered possible null pointers and out of bounds array indexing
- [x] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Comment by 12pm** [comment]
thats nice

File: src/game/server/score.cpp [hunk 1]
```diff
@@ -665,7 +665,11 @@ bool CScore::ShowRankThread(IDbConnection *pSqlServer, const ISqlData *pGameData
 	char aBuf[600];
 
 	str_format(aBuf, sizeof(aBuf),
-		"SELECT Rank, Name, Time "
+		"SELECT Rank, Time, "
+		"  100.0 - 100.0 * (Rank - 1) / ("
+		"    SELECT COUNT(DISTINCT Name) - 1 "
+		"    FROM %s_race WHERE Map = ?"
+		"  ) AS BetterThanPercent "
```
Comment (Zwelf) [inline src/game/server/score.cpp:672]: I would prefer to keep the query simple. Is there a reason to calculate the percentage in the query and not just select the number of ranks or use the PERCENTAGE_RANK window function like so?:

```sql
SELECT Rank, Time, PercentRank
FROM (
  SELECT RANK() OVER w AS Rank, PERCENT_RANK() OVER w as PercentRank, Name, MIN(Time) AS Time
  FROM record_race   WHERE Map = 'solo'
  GROUP BY Name
  WINDOW w AS (ORDER BY Time)
) as a
WHERE Name = 'Zwelf'
```

I don't know if there is a performance difference, but here is the explain output from mysql:
```sql
MariaDB [teeworlds]> explain SELECT Rank, Time,   100.0 - 100.0 * (Rank - 1) / (    SELECT COUNT(DISTINCT Name) - 1     FROM record_race WHERE Map = 'solo'  ) AS BetterThanPercent FROM (  SELECT RANK() OVER w AS Rank, Name, MIN(Time) AS Time   FROM record_race   WHERE Map = 'solo'   GROUP BY Name   WINDOW w AS (ORDER BY Time)) as a WHERE Name = 'Zwelf';
+------+-------------+-------------+------+------------------------+------+---------+-------+------+-------------------------------------------+
| id   | select_type | table       | type | possible_keys          | key  | key_len | ref   | rows | Extra                                     |
+------+-------------+-------------+------+------------------------+------+---------+-------+------+-------------------------------------------+
|    1 | PRIMARY     | <derived3>  | ALL  | NULL                   | NULL | NULL    | NULL  |   13 | Using where                               |
|    3 | DERIVED     | record_race | ref  | Map,Map_2,MapTimestamp | Map  | 514     | const |   13 | Using where; Using index; Using temporary |
|    2 | SUBQUERY    | record_race | ref  | Map,Map_2,MapTimestamp | Map  | 514     | const |   13 | Using where; Using index                  |
+------+-------------+-------------+------+------------------------+------+---------+-------+------+-------------------------------------------+
3 rows in set (0.017 sec)

MariaDB [teeworlds]> explain SELECT Rank, Time, PercentRank FROM (  SELECT RANK() OVER w AS Rank, PERCENT_RANK() OVER w as PercentRank, Name, MIN(Time) AS Time   FROM record_race   WHERE Map = 'solo'   GROUP BY Name   WINDOW w AS (ORDER BY Time)) as a WHERE Name = '[D] Zwelf';
+------+-------------+-------------+------+------------------------+------+---------+-------+------+-------------------------------------------+
| id   | select_type | table       | type | possible_keys          | key  | key_len | ref   | rows | Extra                                     |
+------+-------------+-------------+------+------------------------+------+---------+-------+------+-------------------------------------------+
|    1 | PRIMARY     | <derived2>  | ALL  | NULL                   | NULL | NULL    | NULL  |   13 | Using where                               |
|    2 | DERIVED     | record_race | ref  | Map,Map_2,MapTimestamp | Map  | 514     | const |   13 | Using where; Using index; Using temporary |
+------+-------------+-------------+------+------------------------+------+---------+-------+------+-------------------------------------------+
2 rows in set (0.000 sec)

```

Comment (def-) [reply src/game/server/score.cpp:672]: Nope, that looks fine. You can push another change on top of this?

Comment (Zwelf) [reply src/game/server/score.cpp]: Done
* [x] Tested the change ingame

**Comment by def-** [comment]
bors r+
Thanks!

**Pull request merged**